core/server/master/src/main/java/alluxio/master/block/meta/MasterWorkerInfo.java

```diff
@@ -375,10 +378,19 @@ public boolean hasLostStorage() {
 
   @Override
   public String toString() {
+    Collection<Long> blocks = mBlocks;
+    // We truncate the list of block IDs to print, unless it is for DEBUG logs
+    if (!LOG.isDebugEnabled()) {
+      blocks = (mBlocks.size() < BLOCK_SIZE_LIMIT) ? mBlocks :
+              mBlocks.stream().limit(BLOCK_SIZE_LIMIT).collect(Collectors.toList());
+    }
     return MoreObjects.toStringHelper(this).add("id", mId).add("workerAddress", mWorkerAddress)
-        .add("capacityBytes", mCapacityBytes).add("usedBytes", mUsedBytes)
-        .add("lastUpdatedTimeMs", mLastUpdatedTimeMs).add("blocks", mBlocks)
-        .add("lostStorage", mLostStorage).toString();
+            .add("capacityBytes", mCapacityBytes).add("usedBytes", mUsedBytes)
+            .add("lastUpdatedTimeMs", mLastUpdatedTimeMs).add("blocks", mBlocks)
```

Comment:
This adds `blocks` twice?

Also, if the list was actually truncated, can we label it like `blocks-truncated` or something that distinguishes it from the full blocks list?

Reply:
oops i overlooked that field

Yea makes sense to me `blocks-truncated` would be a better name.